src/connectors/search-box/connectSearchBox.ts

```diff
@@ -140,28 +133,19 @@ const connectSearchBox: SearchBoxConnector = function connectSearchBox(
 
       getWidgetRenderState({ helper, searchMetadata, state }) {
         if (!_refine) {
-          const setQueryAndSearch = (query: string) => {
-            if (query !== state.query) {
```

Comment:
removing this conditional is what really fixes the bug, but the rest of this PR is cleanup

Reply:
I've tested React InstantSearch hooks, Angular InstantSearch, Vue InstantSearch and vanilla and I see no cases where refine gets called with the same query as already was the case, so this previous optimisation isn't needed as far as I can tell

Reply:
For reference, this branch was introduced in 81c2e80dd7ebb1bdac284a579b898c436babc03b.